pyomo/contrib/doe/examples/reactor_compute_FIM.py

```diff
@@ -0,0 +1,105 @@
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#
+#  Pyomo.DoE was produced under the Department of Energy Carbon Capture Simulation
+#  Initiative (CCSI), and is copyright (c) 2022 by the software owners:
+#  TRIAD National Security, LLC., Lawrence Livermore National Security, LLC.,
+#  Lawrence Berkeley National Laboratory, Pacific Northwest National Laboratory,
+#  Battelle Memorial Institute, University of Notre Dame,
+#  The University of Pittsburgh, The University of Texas at Austin,
+#  University of Toledo, West Virginia University, et al. All rights reserved.
+#
+#  NOTICE. This Software was developed under funding from the
+#  U.S. Department of Energy and the U.S. Government consequently retains
+#  certain rights. As such, the U.S. Government has been granted for itself
+#  and others acting on its behalf a paid-up, nonexclusive, irrevocable,
+#  worldwide license in the Software to reproduce, distribute copies to the
+#  public, prepare derivative works, and perform publicly and display
+#  publicly, and to permit other to do so.
+#  ___________________________________________________________________________
+
+
+import numpy as np
+import pyomo.common.unittest as unittest
+from pyomo.contrib.doe.examples.reactor_kinetics import create_model, disc_for_measure
+from pyomo.contrib.doe import (
+    DesignOfExperiments,
+    MeasurementVariables,
+    calculation_mode,
+    DesignVariables,
+    finite_difference_step,
+)
+
+
+def main():
+    ### Define inputs
+    # Control time set [h]
+    t_control = [0, 0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875, 1]
+    # Define parameter nominal value
+    parameter_dict = {'A1': 85, 'A2': 370, 'E1': 8, 'E2': 15}
+
+    # Define measurement object
+    measurements = MeasurementVariables()
+    measurements.add_variables("C", # measurement object
+                               indices={0: ['CA', 'CB', 'CC'], 1: t_control}, # 0,1 are indices of the index sets 
+                               time_index_position=1)
+
+    # design object
+    exp_design = DesignVariables()
+
+    # add CAO as design variable
+    exp_design.add_variables(
+        "CA0",
+        indices={0: [0]},
+        time_index_position=0,
+        values= [5],
+        lower_bounds=1,
+        upper_bounds=5,
+    )
+
+    # add T as design variable
+    exp_design.add_variables(
+        'T',
+        indices= {0: t_control},
+        time_index_position=0,
+        values=[570, 300, 300, 300, 300, 300, 300, 300, 300], # same length with t_control
+        lower_bounds=300,
+        upper_bounds=700,
+    )
+
+    ### Test sequential_finite mode
+    sensi_opt = calculation_mode.sequential_finite
+
+    doe_object = DesignOfExperiments(
+        parameter_dict,
+        exp_design,
+        measurements,
+        create_model,
+        discretize_model=disc_for_measure,
```

Comment:
Add to TODO list to revisit if we still need the discretizer. @blnicho vaguely remembers are example of running the the integrator on an already discretized model. This might be in a PR or development branch.

Reply:
What is the behavior if discretize_model=None? Is this what we do for a non-dynamic model?

Reply:
> What is the behavior if discretize_model=None? Is this what we do for a non-dynamic model?

Yes. If it's a non-dynamic model, we can set discretize_model=None. 

> We should check this explicitly and add a pytest to ensure the correct exception is thrown.

I added this check. A pytest is added to see if the correct exception is thrown.




> Add to TODO list to revisit if we still need the discretizer. @blnicho vaguely remembers are example of running the the integrator on an already discretized model. This might be in a PR or development branch.

Added to the todo list.